plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,79 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Docker struct {
+	ID   string `toml:"id"`
+	Path string `toml:"path"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if d.Path == "" {
+		return errors.New("path missing")
+	}
+	if _, err := os.Stat(d.Path); os.IsNotExist(err) {
+		return errors.New("directory does not exist")
+	}
```

Comment:
Does this directory also exist in docker containers without secrets set?

Reply:
this is tricky, `/run` will always exist in a container however `/run/secrets` will only exist when these Docker Secrets are mentioned and created in the respective `docker-compose.yml` file.

Reply:
This could be a very interesting case. Based on a very simple [demo repo](https://github.com/shantanoo-desai/docker-compose-secrets-envvars) I can confirm that when `secrets` list is not mentioned under the respective service the `/run/secrets/` directory is not created. However, upon inserting the `secrets` list back and recreating the container does create `/run/secrets/` directory.

Reply:
@Hipska I see your use-case and think this is a matter of definition. We could define "if there is no /run/secrets directory that counts as there-are-no-secrets"... I'm not against this case even though `jose` will fail if the `path` does not exist... In any case, I would add a comment to the code documenting this decision and the reasoning behind.

Reply:
So what is the resolution for this review? Should I keep error here and mention comments explicitly as follows:

```go
if _, err := os.Stat(d.Path); os.IsNotExist(err) {
		// if there is no /run/secrets directory for default Path value
		// this implies that there are no secrets.
		// Or for any explicit path definitions for that matter.
		return errors.New("directory does not exist")
}
```